src/components/TodoList.tsx

```diff
@@ -0,0 +1,104 @@
+import classNames from 'classnames';
+import { Todo } from '../types';
+
+type Props = {
+  todos: Todo[];
+  tempTodo: Todo | null;
+  onDelete: (id: number) => void;
+  isLoading: boolean;
+  deletingTodoId: number | null;
+};
+
+export const TodoList: React.FC<Props> = ({
+  todos,
+  tempTodo,
+  onDelete,
+  isLoading,
+  deletingTodoId,
+}) => {
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {todos.map(todo => (
+        <div
+          data-cy="Todo"
+          className={classNames('todo', {
+            completed: todo.completed,
+          })}
+          key={todo.id}
+        >
+          <label className="todo__status-label">
+            <input
+              data-cy="TodoStatus"
+              type="checkbox"
+              className="todo__status"
+              checked={todo.completed}
+            />
+          </label>
+          <span data-cy="TodoTitle" className="todo__title">
+            {todo.title}
+          </span>
+          {/* Remove button appears only on hover */}
+          <button
+            type="button"
+            className="todo__remove"
+            data-cy="TodoDelete"
+            onClick={() => onDelete(todo.id)}
+          >
+            ×
+          </button>
+          {/* overlay will cover the todo while it is being deleted or updated */}
+          <div
+            data-cy="TodoLoader"
+            className={classNames('modal overlay', {
+              'is-active': todo.id === deletingTodoId,
+            })}
+          >
+            <div className="modal-background has-background-white-ter" />
+            <div className="loader" />
+          </div>
+        </div>
+      ))}
+
+      {tempTodo !== null && (
+        <div
+          data-cy="Todo"
+          className={classNames('todo', {
+            completed: tempTodo.completed,
+          })}
+          key={tempTodo.id}
+        >
+          <label className="todo__status-label">
+            <input
+              data-cy="TodoStatus"
+              type="checkbox"
+              className="todo__status"
+              checked={tempTodo.completed}
+            />
+          </label>
+          <span data-cy="TodoTitle" className="todo__title">
+            {tempTodo.title}
+          </span>
+          {/* Remove button appears only on hover */}
+          <button
+            type="button"
+            className="todo__remove"
+            data-cy="TodoDelete"
+            onClick={() => onDelete(tempTodo.id)}
+          >
+            ×
+          </button>
+          {/* overlay will cover the todo while it is being deleted or updated */}
+          <div
+            data-cy="TodoLoader"
+            className={classNames('modal overlay', { 'is-active': isLoading })}
+          >
+            <div className="modal-background has-background-white-ter" />
+            <div className="loader" />
+          </div>
+        </div>
+      )}
+    </section>
```

Comment:
create component TempTodo

Reply:
or use TempTodo component, because of differences in logic and markup